### PR TITLE
timeoutSeconds config in .tbrc.yml

### PR DESCRIPTION
**Pull request by l0rdcafe**

### CODE REVIEW

File: config/config.go
Comment (cszatmary) [inline config/config.go:156]: Does this need to be a pointer? Why not just treat the zero value (literally 0 in this case lol) as meaning no timeout? Based on this validation 0 isn't even allowed as a valid value so why not use it for this?

I think this make make things simpler and this is pretty standard practice in Go. For example, see the `Timeout` field on `http.Client`: https://pkg.go.dev/net/http#Client

> A Timeout of zero means no timeout.

File: engine/engine.go
Comment (cszatmary) [inline engine/engine.go:77]: Why not just make this a `time.Duration`? That's what it's meant to be might as well store it as that. Also means you don't need to convert it every time you use it

Comment (cszatmary) [inline engine/engine.go:77]: Is this only gonna be for docker pulls? Wouldn't we want this to be a general purpose timeout for all operations?

File: config/config.go
Comment (l0rdcafe) [reply config/config.go:156]: that's a good point, though is there a way to know if someone explicitly set timeoutSeconds: 0 or if it's just not provided in .tbrc.yml? i'm asking because we're defaulting to 3600 when the value is not provided and i'm not sure if there is a way to distinguish between it not being set or it being set to 0

File: engine/engine.go
Comment (l0rdcafe) [reply engine/engine.go]: i ended up using it for all operations as well

Comment (l0rdcafe) [reply engine/engine.go]: sounds good, i used `time.Duration` all over but i needed to use `int` for the types used to parse the `.tbrc.yml`. i was getting an unmarshal error for some reason

Comment (cszatmary) [reply engine/engine.go]: yeah it needs to be an int for the config, but for the Engine field it should be a time.Duration. Basically at the config layer we read it as an int but then convert it to a duration and use it as that in the rest of the app

File: config/config.go
Comment (cszatmary) [reply config/config.go]: There isn't any way to distinguish, someone could put `timeoutSeconds: 0` in their `.tbrc.yml` but I guess the question is does it matter? I don't see why anyone would do that, and even if they did I don't think it matters, the behaviour should be the same

File: config/config.go
Comment (cszatmary) [inline config/config.go:167]: Nit, could simplify this a bit by assigning the default along with the declaration and eliminate the else case.

File: config/config.go
Comment (cszatmary) [inline config/config.go:166]: Nit, you don't need to explicitly convert numeric constants, i.e. `3600 * time.Second` works

File: config/config.go
Comment (l0rdcafe) [reply config/config.go]: sounds good, thanks for the suggestion. looks simpler now

File: config/config.go
Comment (cszatmary) [inline config/config.go:162]: Actually, you could have even done `60 * time.Minute` 😄

Comment (l0rdcafe) [reply config/config.go:162]: haha great call, let me do that instead 😄

**Review by cszatmary — APPROVED** [review]
You'll need to get an approval from someone else at TB since I don't have write access anymore but this lgtm, great job!

**Review by ohsabry — APPROVED** [review]

**Pull request merged**
